Services.Test/Storage/StorageRecordsTest.cs

```diff
@@ -211,6 +238,54 @@ public void ItCreatesRecords()
             Assert.Equal("bar", (createTask.Result as StorageRecord).GetDocumentDbRecord().LockOwnerType);
         }
 
+        [Fact]
+        public void ItThrowsConflictingResourceExceptionWhenCreatingAResourceWithAnIdThatIsInUse()
+        {
+            // Arrange
+            var exception = this.BuildDocumentClinetException(HttpStatusCode.Conflict);
+
+            this.mockDocumentDbWrapper.Setup(
+                x => x.CreateAsync(It.IsAny<IDocumentClient>(), It.IsAny<StorageConfig>(), It.IsAny<DocumentDbRecord>())
+            ).ThrowsAsync(
+                (DocumentClientException)exception
+            );
+
+            // Mock storage record
+            var mockStorageRecord = new Mock<StorageRecord>();
+
+            // Act, Assert
+            Assert.ThrowsAnyAsync<ConflictingResourceException>(
+                    () => this.target.CreateAsync(mockStorageRecord.Object)
+                )
+                .Wait(testTimeout);
+        }
+
+        [Fact]
+        public void ItLogsErrorWhenTryingToDeleteARecordThatDoesNotExist()
+        {
+            // Arrange
+            var exception = this.BuildDocumentClinetException(HttpStatusCode.NotFound);
+
+            this.mockDocumentDbWrapper.Setup(
+                x => x.DeleteAsync(It.IsAny<IDocumentClient>(), It.IsAny<StorageConfig>(), It.IsAny<string>())
+            ).ThrowsAsync(
+                (DocumentClientException)exception
+            );
+
+            // Act
+            var deleteTask = this.target.DeleteAsync("123");
+            deleteTask.Wait(testTimeout);
+
+            // Act, Assert
+            this.mockLogger.Verify(
+                x => x.Debug(
+                    It.Is<string>(_ => _.Contains(EXPECTED_DEBUG_LOG_RESOURCE_DOES_NOT_EXIST)),
```

Comment:
please don't do this. Checking the code correctness by checking the content of strings, e.g. error messages, is not very robust. For instance, this test will fail if we remove the "." at the end of the error msg or if we rephrase the msg.

Since this is verifying that a call to `.Debug` is made, it's probably not worth testing this scenario. Btw the name of the test indicates that the app logs an error, but then the test expects a call to .Debug().

Also note, that often we don't test calls to the logger, unless they are critical

Reply:
I changed the test to:
1) verify that Debug is called once
2) renamed to reflect that it logs debug

I think it's worth testing that we log when an invalid ID is used... this feels like an important scenario.